packages/server/src/koaJsonRpc/index.ts

```diff
@@ -110,9 +114,10 @@ export default class KoaJsonRpc {
         return;
       }
 
+      this.requestId = ctx.state.reqId;
       const methodName = body.method;
       const methodTotalLimit = this.registryTotal[methodName];
-      if (this.rateLimit.shouldRateLimit(ctx.ip, methodName, methodTotalLimit)) {
+      if (this.rateLimit.shouldRateLimit(ctx.ip, ctx.state.reqId, methodName, methodTotalLimit)) {
```

Comment:
you could also have used here `this.requestId` (see line 117 just above) - don't know which is better readability-wise.

Reply:
Updated.